### PR TITLE
Add base class for building web services with FastAPI.

### LINKED ISSUE #262: Create parent class for containers
### Why are we doing this?
Our containerized services share a lot of boilerplate code to set up FastAPI. This code is currently replicated across all containerized services, which is not best practice.

### Action Requested
Create a parent class in the `container` folder that sets up this boilerplate code, and edit existing services to inherit from the parent class.

In particular, this base class should set up the `app` instance of `FastAPI` and create the default health check endpoint.

### Acceptance Criteria
Code duplicated at service setup (instantiating FastAPI and the health check, in particular) should live in a single location instead of being replicated across services.

### Additional Information
See this [PR review](https://github.com/CDCgov/phdi/pull/246/files/905bf520ad60fad957d04cc79ccc63dea0de2355#r1130028881) for additional context.

### PR DESCRIPTION
**Pull request by DanPaseltiner**
This PR introduces a base class for the FastAPI services we are using to construct our containerized Building Blocks.

fixes #262

### CODE REVIEW

**Review by BradySkylight — APPROVED** [review]
LGTM!  Nice work.  This is going to be nice to have and make it easier to stand-up containers of our BBs.

File: phdi/containers/base_service.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        """
+        Return a FastAPI instance with DIBBs metadata set. If
+        `include_health_check_endpoint` is True, then the health check endpoint
+        will be added.
+
+        :return: The FastAPI instance.
+        """
+        if self.include_health_check_endpoint:
+            self.add_health_check_endpoint()
+        return self.app
```
[the remark this answers is not in the document]
Comment (DanPaseltiner) [reply phdi/containers/base_service.py:63]: Sure thing! I should have a PR up shortly for a new container that will use this class. Refactoring the existing containers to use this class will need to be some tech debt tickets.

File: phdi/containers/base_service.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            },
+            description=description,
+        )
+
+    def add_health_check_endpoint(self):
+        @self.app.get("/")
+        async def health_check() -> dict:
+            """
+            Check service status. If an HTTP 200 status code is returned along with
+            '{"status": "OK"}' then the tabulation service is available and running
```
Comment (emmastephenson) [inline phdi/containers/base_service.py:48]: Nit: this comment should probably be adjusted to be more generic ("the service is available and running properly")

**Review by emmastephenson — APPROVED** [review]
This looks great, thanks Dan!

File: phdi/containers/base_service.py (hunk 2 above)
Comment (DanPaseltiner) [reply phdi/containers/base_service.py]: updated.

**Pull request merged**